apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -264,17 +264,24 @@ def _bind_dvs_port(self, context, segment, agent):
         """
         if self._check_segment_for_agent(segment, agent):
             network_id = context.current.get('network_id')
-            network_name = self.name_mapper.network(context, network_id)
+            epg_name = self.name_mapper.network(context, network_id)
             net = self._get_plugin().get_network(context._plugin_context,
                                                  network_id)
-            project_name = self.name_mapper.tenant(None, net['tenant_id'])
             # Use default security groups from MD
             vif_details = {portbindings.CAP_PORT_FILTER: self.sg_enabled}
-            vif_details['dvs_port_group_name'] = (cfg.CONF.apic_system_id +
-                                                  '|' + str(project_name) +
-                                                  '|' + str(network_name))
+            if not self.single_tenant_mode:
```

Comment:
Why not use _get_network_aci_tenant() and _get_network_app_profile() in both cases? Both functions account for single_tenant_mode.